### PR TITLE
[ios][android] Drop SDK 33

### LINKED ISSUE #7012: Drop SDK33
For SDK37 release we want to drop SDK33 and no others. This must be done before **February 21.**

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Resolves https://github.com/expo/expo/issues/7012.

# How

Run `et remove-sdk`. 

# Test Plan

| 🛠 |   Android  |   iOS  |
| --- | :---: | :---: |
| NCL | ✅ | ✅ |

### CODE REVIEW

File: android/.idea/codeStyles/codeStyleConfig.xml [hunk 1]
```diff
@@ -1,5 +0,0 @@
-<component name="ProjectCodeStyleConfiguration">
```
Comment (bbarthec) [inline android/.idea/codeStyles/codeStyleConfig.xml]: As I see more and more changes to this set of files I'd like to have it settled once and for a while 😉 I suggest changing this set of rules in separate PR.

File: android/.idea/compiler.xml [hunk 2]
```diff
@@ -1,22 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bbarthec) [inline android/.idea/compiler.xml]: and heree

File: android/.idea/copyright/profiles_settings.xml [hunk 3]
```diff
@@ -1,7 +0,0 @@
-<component name="CopyrightManager">
```
Comment (bbarthec) [inline android/.idea/copyright/profiles_settings.xml]: and here

File: android/.idea/encodings.xml [hunk 4]
```diff
@@ -1,6 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bbarthec) [inline android/.idea/encodings.xml]: and here

File: android/app/build.gradle [hunk 5]
```diff
@@ -221,7 +217,7 @@ dependencies {
 
   /* UNCOMMENT WHEN DISTRIBUTING
   api 'org.webkit:android-jsc:r245459' // needs to be before react-native
-  api 'com.facebook.react:react-native:33.0.0'
+  api 'com.facebook.react:react-native:+'
```
Comment (bbarthec) [inline android/app/build.gradle]: Shouldn't it be 🤔 
```suggestion
  api 'com.facebook.react:react-native:34.0.0'
```

File: android/sdkVersions.json [hunk 6]
```diff
@@ -1 +1 @@
-{"sdkVersions":["33.0.0","34.0.0","35.0.0"]}
\ No newline at end of file
+{"sdkVersions":["34.0.0","35.0.0"]}
```
Comment (bbarthec) [inline android/sdkVersions.json]: That's interesting that this file does not contain `36.0.0` that we're supporting 🤔 cc @tsapeta

**Comment by bbarthec** [comment]
Have you checked for any obsolete code that might be removable at this point, like `remove once SDK33 is phased out`?

File: android/app/build.gradle (hunk 5 above)
Comment (lukmccall) [reply android/app/build.gradle]: I've talked to him about it. 
He suggested changing it to `+`.

**Comment by lukmccall** [comment]
> Have you checked for any obsolete code that might be removable at this point, like `remove once SDK33 is phased out`?

Yes, I've checked it. There wasn't any code we should remove connected with the SDK 33.

File: android/app/build.gradle (hunk 5 above)
Comment (tsapeta) [reply android/app/build.gradle]: @bbarthec why 34? 😄 it's actually our fault that we haven't been updating this when droping previous SDKs and it's just a coincidence it worked.

File: android/sdkVersions.json (hunk 6 above)
Comment (tsapeta) [reply android/sdkVersions.json]: 🤦‍♂ 
Could you @lukmccall check whether it's being updated by `et add-sdk`?

**Comment by tsapeta** [comment]
@lukmccall In terms of the first place in `Kernel.java` then I would ask @esamelson what to do, just to make sure we don't break anything.
Looks like this thing in `FileSystemModule.java` refers to _Android_ SDK 25 that we're still supporting 😄

**Comment by lukmccall** [comment]
> Looks like this thing in FileSystemModule.java refers to Android SDK 25 that we're still supporting 😄
Oh... I see. My bad.

**Review by tsapeta — APPROVED** [review]

**Comment by esamelson** [comment]
that particular codepath in Kernel.java should really never be called, so it shouldn't hurt to remove it (it's unsupported behavior) but it also doesn't break any supported behavior to leave it, so whatever you went with is fine 🤷‍♂

**Pull request merged**